clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -180,4 +213,13 @@ public Node leastLoadedNode(long now) {
         return this.node;
     }
 
+    /**
+     * Request predicates give the tester the ability to set an expectation on the request that
```

Comment:
Yeah, that might be better. Do you think the name RequestPredicate makes the intended usage clear? I was also thinking of calling it RequestMatcher or RequestAssertion.